include/bluetooth/mesh/cfg_cli.h

```diff
@@ -936,6 +940,97 @@ int32_t bt_mesh_cfg_cli_timeout_get(void);
  */
 void bt_mesh_cfg_cli_timeout_set(int32_t timeout);
 
+/** Parsed Composition data page 0 representation.
+ *
+ *  Should be pulled from the return buffer passed to
+ *  @ref bt_mesh_cfg_comp_data_get using
+ *  @ref bt_mesh_comp_page_0_get.
+ */
+struct bt_mesh_comp_page_0 {
+	/** Company ID */
+	uint16_t cid;
+	/** Product ID */
+	uint16_t pid;
+	/** Version ID */
+	uint16_t vid;
+	/** Replay protection list size */
+	uint16_t crpl;
+	/** Supported features, see @ref BT_MESH_FEAT_SUPPORTED. */
+	uint16_t feat;
+
+	struct net_buf_simple *_buf;
+};
+
+/** Composition data page 0 element representation */
+struct bt_mesh_comp_elem {
+	/** Element location */
+	uint16_t loc;
+	/** The number of SIG models in this element */
+	size_t nsig;
+	/** The number of vendor models in this element */
+	size_t nvnd;
+
+	uint8_t *_buf;
+};
+
+/** @brief Create a composition data page 0 representation from a buffer.
+ *
+ *  The composition data page object will take ownership over the buffer, which
+ *  should not be manipulated directly after this call.
+ *
+ *  This function can be used in combination with @ref bt_mesh_cfg_comp_data_get
+ *  to read out composition data page 0 from other devices:
+ *
+ *  @code
+ *  NET_BUF_SIMPLE_DEFINE(buf, BT_MESH_RX_SDU_MAX);
+ *  struct bt_mesh_com_page_0 comp;
+ *
+ *  err = bt_mesh_cfg_comp_data_get(net_idx, addr, 0, &status, &buf);
+ *  if (!err) {
+ *          bt_mesh_comp_page_0_get(&buf, &comp);
+ *  }
+ *  @endcode
+ *
+ *  @param buf  Network buffer containing composition data.
+ *  @param comp Composition data structure to fill.
+ *
+ *  @return 0 on success, or (negative) error code on failure.
+ */
+int bt_mesh_comp_page_0_get(struct bt_mesh_comp_page_0 *comp,
+			    struct net_buf_simple *buf);
+
+/** @brief Pull an element from a composition data page.
```

Comment:
from a composition data page 0?

More instances in this function API description.

If I'm following this right.